### PR TITLE
Add roundtrip marshal tests and fix JSON/YAML tag compatibility for v0mimir1 configs

### PR DESCRIPTION
**Pull request by yuri-tceretian**
## Summary

- Add `GetFullConfig` test helpers for each v0mimir1 integration to enable roundtrip testing
- Add roundtrip marshal/unmarshal tests for integration configs (YAML and JSON)
- Move custom secrets JSON marshaler to `receivers/` package, supporting both upstream and local secret types (`Secret`, `SecretURL`)
- Copy `SigV4Config` struct from upstream into `receivers/sns/v0mimir1` to control JSON/YAML tags
- Fix JSON tag compatibility: Jira `custom_fields`, Telegram `token`/`chat` fallback, SigV4 field tags
- Flatten `Headers` type to remove custom JSON marshaler that interfered with secret unmasking
- Move HTTP client test data to a separate `v0mimir1test` package
- Add `HTTPClientConfig` roundtrip tests documenting known JSON round-trip limitations

## Test plan

- [x] Per-integration config roundtrip tests (YAML and JSON marshal → unmarshal)
- [x] Secret type marshaling tests (masking in standard JSON, plain text in `PlainSecretsJSON`)
- [x] Receiver conversion roundtrip: `definition.Receiver` → upstream `config.Receiver` → `definition.Receiver`
- [x] `HTTPClientConfig` roundtrip tests documenting edge cases
- [x] CI: `make test` and `make lint` pass

🤖 Generated with [Claude Code](https://claude.ai/code)

---

> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit ee2751344d5922c664870e53fd61e5f3e62f8353. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: receivers/telegram/v0mimir1/config.go [hunk 1]
```diff
@@ -52,11 +52,23 @@ type Config struct {
 
 // UnmarshalYAML implements the yaml.Unmarshaler interface.
 func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	*c = DefaultConfig
 	type plain Config
-	if err := unmarshal((*plain)(c)); err != nil {
+	type withFallback struct {
+		plain        `yaml:",inline" json:",inline"`
+		BotTokenJson receivers.Secret `yaml:"token"`
+		ChatIDJson   int64            `yaml:"chat"`
```
[the remark this answers is not in the document]
Comment (yuri-tceretian) [reply receivers/telegram/v0mimir1/config.go:59]: file fields are not supported in Grafana\Mimir. We can ignore them

File: receivers/telegram/v0mimir1/config.go [hunk 2]
```diff
@@ -52,11 +52,23 @@ type Config struct {
 
 // UnmarshalYAML implements the yaml.Unmarshaler interface.
 func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
-	*c = DefaultConfig
 	type plain Config
-	if err := unmarshal((*plain)(c)); err != nil {
+	type withFallback struct {
```
Comment (JacobsonMT) [inline receivers/telegram/v0mimir1/config.go:56]: Might want to leave a comment on why these fallbacks exist here and in JIRA for posterity.

File: receivers/email/v0mimir1/testing.go [hunk 3]
```diff
@@ -14,8 +19,33 @@ const FullValidConfigForTesting = `{
...
-		"insecure_skip_verify": false,     
+		"insecure_skip_verify": false,
 		"server_name": "test-server-name"
 	},
 	"send_resolved": true
 }`
+
+// GetFullValidConfig returns a fully populated Config struct with all fields
+// set to non-zero values.
+func GetFullValidConfig() Config {
```
Comment (JacobsonMT) [inline receivers/email/v0mimir1/testing.go:30]: Why not create `FullValidConfig` from `FullValidConfigForTesting` for each notifier, or vice versa? Right now they both claim to be a means to get a "Fully defined config" but they differ in which fields they populate.

Example here where `Hello`, `HTML`, and some `TLSConfig` exist on one but not the other.

Some alternatives could be to ensure they align with a test checking `ConfigReceiverToMimirIntegrations` against `AllValidMimirConfigs` or just generating the json string in a snapshot test.

**Review by JacobsonMT — APPROVED** [review]
LGTM with a few nits

File: receivers/email/v0mimir1/testing.go (hunk 3 above)
Comment (yuri-tceretian) [reply receivers/email/v0mimir1/testing.go:30]: Yea, I did not want to touch the current jsons of full configs. I decided that for roundtrip tests they would not work (to unreliable to count on json unmarshal to re-marshal). I think I will replace the serialized configs with new ones in the follow up PR
